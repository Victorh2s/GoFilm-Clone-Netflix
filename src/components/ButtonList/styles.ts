import styled, { css } from 'styled-components';

export const button2 = styled.button`
  ${({ theme }) => css`

  border: none;
  background: rgba(255,255,255, 10%);
  padding: 0.5rem 2rem;
  margin-top: 3rem;
  cursor: pointer;
  transition: all 200ms ease-in-out;
  font-weight: bold;
  font-size:  ${theme.font.sizes.small};


  :hover{
    background: rgba(255,255,255, 20%);
  }

  `}
`;

export const link2 = styled.a`
  ${({ theme }) => css`
  display: flex;
  text-decoration: none;
  justify-content: center;
  align-items: center;

>svg{
  width: 4rem;
  color: ${theme.colors.white};
  padding: 0 0.5rem;
}


  `}
`;
