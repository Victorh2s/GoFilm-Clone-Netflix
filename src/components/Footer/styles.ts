import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.dark};
    height: 10rem;

  `}
`;
export const Container = styled.section`
  ${({ theme }) => css`

    display: flex;
    text-align: center;
    justify-content: center;
    padding: ${theme.spacings.small};


  `}
`;

export const link = styled.a`
  ${({ theme }) => css`

  padding:${theme.spacings.small};

  >svg{
    width: 3rem;
    height: 4rem;
    color: ${theme.colors.white};
  }

  `}
`;
