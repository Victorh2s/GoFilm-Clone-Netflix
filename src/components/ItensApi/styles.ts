import styled, { css } from 'styled-components';

export const itens = styled.div`
  ${({ theme }) => css`

  font-size: 1.8rem;
  font-weight: bold;
  margin-top: 1.5rem;
  color: ${theme.colors.white};

  `}
`;

export const Vote = styled.div`
  ${({ theme }) => css`
  display: inline-block;
  color: ${theme.colors.green};
  margin-right: 1.5rem;


  `}
`;

export const Year = styled.div`
  display: inline-block;
  margin-right: 1.5rem;
`;

export const Season = styled.div`
  display: inline-block;
`;

export const Description = styled.div`
  ${({ theme }) => css`

  margin-top: 1.5rem;
  font-size: ${theme.font.sizes.small};
  color: ${theme.colors.mediumGray};
  max-width: 35%;


  `}
`;

export const Genres = styled.div`
  ${({ theme }) => css`

  margin-top: 1.3rem;
  color: ${theme.colors.mediumGray};
  font-size: ${theme.font.sizes.small};
  font-style: italic;

  `}
`;
