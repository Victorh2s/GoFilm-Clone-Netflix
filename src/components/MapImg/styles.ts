import styled from 'styled-components';

export const MovieFilm = styled.div`
  margin-left: 1rem;

`;

export const Link = styled.a`

`;

export const Image = styled.img`
  height: 25rem;
  width: 19rem;
  transform: scale(0.9);
  transition: all 200ms ease-in-out;
  cursor: pointer;
  :hover{
    transform: scale(1);
  }


`;
