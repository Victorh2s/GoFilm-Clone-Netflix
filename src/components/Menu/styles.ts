import styled, { css } from 'styled-components';

export const Menu = styled.menu`
  ${({ theme }) => css`

  display: flex;
  height: 6rem;
  top: 0;
  left:0;
  right: 0;
  background: transparent;
  align-items: center;
  justify-content: space-between;
  overflow: hidden;
  position: fixed;
  transition: all 200ms ease-in-out;
  z-index: 4;



    `}
`;

export const Logo = styled.a`
  padding: 5rem;
`;

export const Netflix = styled.img`
  ${({ theme }) => css`
  width: 12rem;
  height: 12rem;
  `}
`;

export const Avatar = styled.a`
  padding: 5rem;
`;

export const icon = styled.img`
  ${({ theme }) => css`
  width: 3rem;
  height: 3rem;
  `}
`;
