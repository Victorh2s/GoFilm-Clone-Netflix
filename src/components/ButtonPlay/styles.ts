import { Wrapper as Text } from '../Text/styles';
import styled, { css } from 'styled-components';

export const button = styled.button`
  ${({ theme }) => css`

  border: none;
  background: rgba(255,255,255, 70%);
  padding: 0.5rem 2rem;
  margin-top: 3rem;
  cursor: pointer;
  transition: all 200ms ease-in-out;
  font-weight: bold;
  font-size:  ${theme.font.sizes.small};

  :hover{
    background: rgba(255,255,255, 80%);
  }

  `}
`;

export const link = styled.a`
  ${({ theme }) => css`
  text-decoration: none;
  justify-content: center;
  align-items: center;
  display: flex;

  ${Text}{
    color: ${theme.colors.dark};
  }

>svg{
  width: 4rem;
  color: black;
  padding: 0 0.5rem;
}
text-decoration: none;
color: ${theme.colors.dark};

  `}
`;
