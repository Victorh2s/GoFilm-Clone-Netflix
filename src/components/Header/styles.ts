import styled from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Wrapper = styled.section`
    height: 97vh;
`;

export const GladientVertical = styled.div`
width: inherit;
height:inherit;
background: linear-gradient(to top, #0d0c0c 5%, transparent 100%);

`;

export const GladientHorizontal = styled.div`

width: inherit;
height:inherit;
background: linear-gradient(to right , #0d0c0c 18%, transparent 95%);
display: flex;
flex-direction: column;
justify-content: center;
padding-left: 3rem;
padding-top: 4rem;

${Heading}{
  max-width: 50%;
}

`;
