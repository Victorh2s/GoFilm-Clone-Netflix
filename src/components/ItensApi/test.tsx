import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ItensApi, ItensApiProps } from '.';

const props: ItensApiProps = {
  id: 1,
  vote: 9.5,
  year: new Date('2016-07-15'),
  seasons: 5,
  description: 'Descrição',
  genresName: ['terror', 'comédia'],
};

describe('<ItensApi />', () => {
  it('should render Vote, Year, seasons, description, genresName', () => {
    renderTheme(<ItensApi {...props} />);

    expect(screen.getByText('9.5 pontos')).toBeInTheDocument();
    expect(screen.getByText('9.5 pontos')).toHaveStyle({
      display: 'inline-block',
    });
    expect(screen.getByText(/Ano 2016/i)).toBeInTheDocument();
    expect(screen.getByText(/Ano 2016/i)).toHaveStyle({
      display: 'inline-block',
    });
    expect(screen.getByText(/5 temporadas/i)).toBeInTheDocument();
    expect(screen.getByText(/Descrição/i)).toBeInTheDocument();
    expect(screen.getByText(/terror, comédia/i)).toBeInTheDocument();
    expect(screen.getByText(/Assistir/i)).toBeInTheDocument();
    expect(screen.getByText(/Minha Lista/i)).toBeInTheDocument();
  });

  it('should match to snapshot', () => {
    const { container } = renderTheme(<ItensApi {...props} />);

    expect(container).toMatchSnapshot();
  });
});
