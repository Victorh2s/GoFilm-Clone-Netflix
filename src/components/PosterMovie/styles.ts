import { Description } from '../ItensApi/styles';
import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`

  background: ${theme.colors.dark};
  min-height: 100vh;
  display: flex;
  margin-top: 3rem;


  @media ${theme.media.lteSmall} {
    display: block;

    ${Image}{
      margin: 0;
    }

    ${YearAndSeason}{
      margin-top: -5rem;
    }



  }
  @media ${theme.media.lteSuperSmall} {
    display: block;

    ${Image}{
      margin: 0;
        max-width: 45rem;
    }

    ${YearAndSeason}{
      margin-top: -5rem;
    }



  }

  `}
`;

export const Poster = styled.div`
  text-align: center;
  justify-content: center;

`;

export const Image = styled.img`
  ${({ theme }) => css`


  max-width: 45rem;
  padding: ${theme.spacings.large};
  margin-left: 5rem;



  `}
`;

export const YearAndSeason = styled.section`
  ${({ theme }) => css`

    padding-top: ${theme.spacings.xsmall};
    padding: ${theme.spacings.xsmall};


  ${Description}{
    max-width: 100%;
  }


  `}
`;
