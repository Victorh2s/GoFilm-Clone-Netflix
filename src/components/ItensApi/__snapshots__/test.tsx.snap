// Vitest Snapshot v1

exports[`<ItensApi /> > should match to snapshot 1`] = `
.c10 {
  font-size: 1.6rem;
  color: #FFFFFF;
  font-family: 'Open Sans',sans-serif;
}

.c10 a,
.c10 a:visited,
.c10 a:link {
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c10 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c11 {
  border: none;
  background: rgba(255,255,255,10%);
  padding: 0.5rem 2rem;
  margin-top: 3rem;
  cursor: pointer;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  font-weight: bold;
  font-size: 1.6rem;
}

.c11:hover {
  background: rgba(255,255,255,20%);
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 >svg {
  width: 4rem;
  color: #FFFFFF;
  padding: 0 0.5rem;
}

.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  border: none;
  background: rgba(255,255,255,70%);
  padding: 0.5rem 2rem;
  margin-top: 3rem;
  cursor: pointer;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  font-weight: bold;
  font-size: 1.6rem;
}

.c6:hover {
  background: rgba(255,255,255,80%);
}

.c7 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0d0c0c;
}

.c7 .c9 {
  color: #0d0c0c;
}

.c7 >svg {
  width: 4rem;
  color: black;
  padding: 0 0.5rem;
}

.c0 {
  font-size: 1.8rem;
  font-weight: bold;
  margin-top: 1.5rem;
  color: #FFFFFF;
}

.c1 {
  display: inline-block;
  color: #46d369;
  margin-right: 1.5rem;
}

.c2 {
  display: inline-block;
  margin-right: 1.5rem;
}

.c3 {
  display: inline-block;
}

.c4 {
  margin-top: 1.5rem;
  font-size: 1.6rem;
  color: #C6C8CC;
  max-width: 35%;
}

.c5 {
  margin-top: 1.3rem;
  color: #C6C8CC;
  font-size: 1.6rem;
  font-style: italic;
}

@media (max-width:768px) {
  .c10 {
    font-size: 1.6rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
       
      9.5
       pontos
    </div>
    <div
      class="c2"
    >
       
      2016
    </div>
    <div
      class="c3"
    >
      5 temporada
      s
    </div>
    <div
      class="c4"
    >
      Descrição
    </div>
    <div
      class="c5"
    >
      <strong>
        Gêneros: 
      </strong>
      terror, comédia
    </div>
    <button
      class="c6"
    >
      <a
        class="c7"
        href="/watch/1"
      >
        <svg
          aria-hidden="true"
          class="c8"
          fill="currentColor"
          focusable="false"
          role="Play"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.804 8 5 4.633v6.734L10.804 8zm.792-.696a.802.802 0 0 1 0 1.392l-6.363 3.692C4.713 12.69 4 12.345 4 11.692V4.308c0-.653.713-.998 1.233-.696l6.363 3.692z"
          />
        </svg>
         
        <div
          class="c9 c10"
        >
          Assistir
        </div>
      </a>
    </button>
    <button
      class="c11"
    >
      <a
        class="c12"
        href="list/add/1"
      >
        <svg
          aria-hidden="true"
          class="c8"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.4 9H16V5.6c0-.6-.4-.6-1-.6s-1 0-1 .6V9h-3.4c-.6 0-.6.4-.6 1s0 1 .6 1H14v3.4c0 .6.4.6 1 .6s1 0 1-.6V11h3.4c.6 0 .6-.4.6-1s0-1-.6-1zm-12 0H.6C0 9 0 9.4 0 10s0 1 .6 1h6.8c.6 0 .6-.4.6-1s0-1-.6-1zm0 5H.6c-.6 0-.6.4-.6 1s0 1 .6 1h6.8c.6 0 .6-.4.6-1s0-1-.6-1zm0-10H.6C0 4 0 4.4 0 5s0 1 .6 1h6.8C8 6 8 5.6 8 5s0-1-.6-1z"
          />
        </svg>
         
        <div
          class="c9 c10"
        >
          Minha Lista
        </div>
      </a>
    </button>
  </div>
</div>
`;
